### PR TITLE
Changed the type of cell style from string to CellStyle.

### PR DESCRIPTION
**Pull request by junsikshim**
**Summary**

Previously when inserting a vertex or an edge into a graph, a string-typed value was needed to style it.
```js
// Before
graph.insertVertex({
  ...
  style: 'shape=cylinder;strokeWidth=2;fillColor:#ffffff'
});
```

This has brought unnecessary parsing and concating string operations, and also hindered code readability with no syntax highlighting to support it.
Now an object is used instead to replace the string, so that the previous example becomes:
```js
// Now
graph.insertVertex({
  ...
  style: {
    shape: 'cylinder',
    strokeWidth: 2,
    fillColor: '#ffffff'
  }
});
```

A new type called `CellStyle` is introduced to represent the style object, and it is almost identical to `CellStateStyle`.
```js
export type CellStyle = CellStateStyle & { baseStyleName?: string };
```

I've added a new property named `baseStyleName` which replaces the previous `stylename`, so that,
```js
style: 'top'
```
becomes,
```js
style: {
  baseStyleName: 'top'
}
```

In order to cover the case when the style starts with a semicolon(without any default style), a `null` can be assigned to `baseStyleName` to explicitly unset it.
```js
style: {
  baseStyleName: null
}
```

I've updated all the examples accordingly, but I'm not 100% sure that everything works.
I'll have to work on the examples one by one after this PR gets merged.

Thanks!

**Description for the changelog**
Changed the type of cell style from string to CellStyle.

**Other info**

### CODE REVIEW

File: packages/core/src/view/mixins/EdgeMixin.ts [hunk 1]
```diff
@@ -189,7 +190,7 @@ const EdgeMixin: PartialType = {
    * on an edge is being double clicked.
    * @default null
```
Comment (tbouffard) [inline packages/core/src/view/mixins/EdgeMixin.ts]: default value is no more `null` 😄

File: packages/core/src/view/style/Stylesheet.ts [hunk 2]
```diff
@@ -178,60 +175,36 @@ export class Stylesheet {
...
-      if (style && name.charAt(0) !== ';') {
-        style = clone(style);
-      } else {
-        style = {} as CellStateStyles;
-      }
-
-      // Parses each key, value pair into the existing style
-      for (const tmp of pairs) {
-        const pos = tmp.indexOf('=');
+  getCellStyle(cellStyle: CellStyle, defaultStyle: CellStateStyle) {
```
Comment (tbouffard) [inline packages/core/src/view/style/Stylesheet.ts:189]: ❓ can we add the return type in the method signature for consistency with other methods?

File: packages/core/src/view/style/Stylesheet.ts [hunk 3]
```diff
@@ -178,60 +175,36 @@ export class Stylesheet {
...
+    };
 
-          if (value === NONE) {
-            delete style[key];
-          } else if (isNumeric(value)) {
-            (style[key] as any) = parseFloat(value);
-          } else {
-            (style[key] as any) = value;
-          }
-        } else {
```
Comment (tbouffard) [inline packages/core/src/view/style/Stylesheet.ts:222]: We are loosing something in the new implementation.

Previously, it was possible to provide several style names according to this implementation. It wasn't explicit in the function documentation. This allowed merging various styles.
By convention, the style names were generally put at the beginning of the string, for instance `style1;style2;style3;key1=value1;key2=value2;...` but this was not mandatory (the extra style names can be put anywhere in the string).
Clearly, we miss unit tests here that would have exhibit the existing behavior 👿 .

This is something [I rely on today with mxgraph](https://github.com/process-analytics/bpmn-visualization-js/blob/v0.23.1/test/unit/component/mxgraph/renderer/StyleComputer.test.ts#L177) and that will disappear with the new implementation.
For the record, here is how it was implemented in mxgraph@4.2.2: https://github.com/jgraph/mxgraph/blob/v4.2.2/javascript/src/js/view/mxStylesheet.js#L251.

Could we change the baseStyleName to make it a string or an array of strings and update the implementation of the `getCellStyle` to support it?

**Review by tbouffard — COMMENTED** [review]
Sorry for the late review, I had to do extra checks to confirm that we broke an existing behavior with the new implementation.
I totally approve the move from string to object to manage the style. This also makes things easier when configuring a cell.

File: packages/core/src/view/style/Stylesheet.ts (hunk 3 above)
Comment (junsikshim) [reply packages/core/src/view/style/Stylesheet.ts:222]: Oh, I didn't know it was possible to have multiple style names.
Making it an array of strings and having a left-to-right merge rule seems good.
I'll work on it as soon as possible.
Thanks!

File: packages/core/src/view/style/Stylesheet.ts (hunk 2 above)
Comment (junsikshim) [reply packages/core/src/view/style/Stylesheet.ts:189]: I've been removing the return types in the method signature when IDEs didn't have problem figuring out the actual type.
But, as you said, `getCellStyle` didn't have a clear return type, so I added below for IDEs to correctly figure it out.
```js
let style: CellStateStyle;
```

File: packages/core/src/view/style/Stylesheet.ts (hunk 3 above)
Comment (junsikshim) [reply packages/core/src/view/style/Stylesheet.ts:222]: So, I renamed`baseStyleName` to `baseStyleNames` and changed it to an array of strings.
Accordingly, passing `[]` insteand of `null` clears the default styles.
Hope it works now!

File: packages/core/src/view/style/Stylesheet.ts (hunk 2 above)
Comment (tbouffard) [reply packages/core/src/view/style/Stylesheet.ts:189]: ok, thanks for the information.

**Review by tbouffard — APPROVED** [review]
Sorry for the late review, that's look good.
Thanks for this great improvements

**Pull request merged**